editor/index.js

```diff
@@ -49,17 +50,29 @@ window.jQuery( document ).on( 'heartbeat-tick', ( event, response ) => {
  * The return value of this function is not necessary if we change where we
  * call createEditorInstance(). This is due to metaBox timing.
  *
- * @param {String}  id       Unique identifier for editor instance
- * @param {Object}  post     API entity for post to edit
- * @param {?Object} settings Editor settings object
+ * @param {String}  id           Unique identifier for editor instance
+ * @param {Object}  post         API entity for post to edit
+ * @param {?Object} settings     Editor settings object
+ * @param {?Object} initialState Initial editor state to hydrate
  * @return {Object} Editor interface. Currently supports metabox initialization.
  */
-export function createEditorInstance( id, post, settings ) {
+export function createEditorInstance( id, post, settings, initialState ) {
```

Comment:
It's a bit confusing for me when I look at it for the first time. At the moment we pass `post` which gets later injected into Redux state, `settings` which are passed directly to the `EditorProvider` and then optional `initialState` which again operates on Redux state. My question is if it would make sense to provide a different handler for rebooting. It will increase the number of lines and introduce some code duplication, but it might improve readability here. What do you think?